File: backend/src/modules/leads/lead.model.js
const mongoose = require('mongoose');

const leadSchema = new mongoose.Schema({
  studentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  accommodation: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Accommodation',
    required: true,
  },
  type: {
    type: String,
    enum: ['visit', 'token_booking'],
    default: 'visit',
  },
  status: {
    type: String,
    enum: ['pending', 'confirmed', 'cancelled', 'completed', 'pending_owner_approval'],
    default: 'pending',
  },
  source: {
    type: String,
    enum: ['web', 'app', 'referral', 'organic'],
    default: 'app',
  },
  payment: {
    amount: Number,
    transactionId: String,
    status: {
      type: String,
      enum: ['pending', 'success', 'failed', 'refunded'],
      default: 'pending',
    }
  },
  preferredDate: Date,
  notes: String,
}, {
  timestamps: true,
});

// Index for fetching user's bookings quickly
leadSchema.index({ studentId: 1, createdAt: -1 });

module.exports = mongoose.model('Lead', leadSchema);

File: mobile/lib/features/leads/domain/lead_model.dart
class Lead {
  final String id;
  final String user;
  final String accommodation;
  final String preferredDate;
  final String? notes;
  final String status;

  Lead({
    required this.id,
    required this.user,
    required this.accommodation,
    required this.preferredDate,
    this.notes,
    required this.status,
  });

  factory Lead.fromJson(Map<String, dynamic> json) {
    return Lead(
      id: json['_id'] ?? '',
      user: json['studentId'] ?? '',
      accommodation: (json['accommodation'] is Map) 
          ? (json['accommodation']['name'] ?? '') 
          : (json['accommodation'] ?? ''),
      preferredDate: json['preferredDate'] ?? '',
      notes: json['notes'],
      status: json['status'] ?? 'pending',
    );
  }
}
